jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/ConfigureLambdaDialog.kt

```diff
@@ -0,0 +1,146 @@
+            return ValidationInfo(message("sqs.configure.lambda.validation.function"), view.lambdaFunction)
+        }
+        return null
+    }
+
+    override fun doOKAction() {
+        if (!isOKActionEnabled) {
+            return
+        }
+
+        isOKActionEnabled = false
+        setOKButtonText(message("sqs.configure.lambda.in_progress"))
+
+        launch {
+            try {
+                configureLambda(functionSelected())
+                runInEdt(ModalityState.any()) {
+                    close(OK_EXIT_CODE)
+                }
+                notifyInfo(message("sqs.service_name"), message("sqs.configure.lambda.success", functionSelected()), project)
+            } catch (e: InvalidParameterValueException) {
+                // Exception thrown for invalid permission
+                runInEdt(ModalityState.any()) {
+                    if (ConfirmIamPolicyDialog(project, iamClient, lambdaClient, functionSelected(), queue, view.component).showAndGet()) {
+                        retryConfiguration(functionSelected())
+                    } else {
+                        setOKButtonText(message("general.configure_button"))
+                        isOKActionEnabled = true
+                    }
+                }
+            } catch (e: Exception) {
+                LOG.warn(e) { message("sqs.configure.lambda.error", functionSelected()) }
+                setErrorText(e.message)
+                setOKButtonText(message("general.configure_button"))
+                isOKActionEnabled = true
+            }
+        }
+    }
+
+    private fun functionSelected(): String = view.lambdaFunction.selected()?.functionName() ?: ""
+
+    internal fun configureLambda(functionName: String) {
+        lambdaClient.createEventSourceMapping {
+            it.functionName(functionName)
+            it.eventSourceArn(queue.arn)
+        }
+    }
+
+    // It takes a few seconds for the role policy to update, so this function will attempt configuration for a duration of time until it succeeds.
+    internal suspend fun waitUntilConfigured(functionName: String): String? {
+        var identifier: String? = null
+        try {
+            waitUntil(
+                succeedOn = {
+                    it.eventSourceArn().isNotEmpty()
+                },
+                exceptionsToIgnore = setOf(InvalidParameterValueException::class),
+                exceptionsToStopOn = setOf(LambdaException::class, ResourceConflictException::class, ResourceNotFoundException::class, ServiceException::class),
+                maxDuration = Duration.ofSeconds(CONFIGURATION_WAIT_TIME),
+                call = {
+                    lambdaClient.createEventSourceMapping {
+                        it.functionName(functionName)
+                        it.eventSourceArn(queue.arn)
+                    }.apply {
+                        identifier = this.uuid()
+                    }
+                }
+            )
+        } catch (e: Exception) {
```

Comment:
We should catch `WaiterTimeoutException` explicitly, and otherwise log the exception.